packages/core/src/utils/path.ts

```diff
@@ -129,6 +129,57 @@ export function setInPath(object: NestedRecord, path: string, value: unknown): v
   }
 }
 
+export function setTouchedInPath(object: NestedRecord, path: string, value: boolean): void {
+  if (isEscapedPath(path)) {
+    const cleanPath = cleanupNonNestedPath(path);
+
+    // Set the main path
+    object[cleanPath] = value;
+
+    // Find and set all nested paths that start with this path
+    for (const key in object) {
+      if (key.startsWith(`${cleanPath}.`)) {
+        object[key] = value;
+      }
+    }
+    return;
+  }
```

Comment:
I'm not sure whether this is an issue or not, but I don't believe its possible to mix `[fieldName]` syntax with `dot notation` ?

Reply:
Unsure, but I want to say yes, they wouldn't mix. I will refactor the setter a little bit so that we don't duplicate as much code here.